pkg/pool-manager/virtualmachine_pool.go

```diff
@@ -456,21 +486,21 @@ func (p *PoolManager) vmWaitingCleanupLook(waitTime int) {
 			t, err := time.Parse(time.RFC3339, allocationTime)
 			if err != nil {
 				// TODO: remove the mac from the wait map??
-				log.Error(err, "failed to parse allocation time")
+				logger.Error(err, "failed to parse allocation time")
 				continue
 			}
 
 			if time.Now().After(t.Add(time.Duration(waitTime) * time.Second)) {
+				configMapUpdateNeeded = true
 				delete(configMap.Data, macAddress)
 				macAddress = strings.Replace(macAddress, "-", ":", 5)
 				delete(p.macPoolMap, macAddress)
-				log.V(1).Info("released mac address in waiting loop", "macAddress", macAddress)
+				logger.V(1).Info("released mac address in waiting loop", "macAddress", macAddress)
 			}
 		}
 
-		_, err = p.kubeClient.CoreV1().ConfigMaps(p.managerNamespace).Update(configMap)
-		if err != nil {
-			log.Error(err, "failed to update config map", "configMapName", names.WAITING_VMS_CONFIGMAP)
+		if configMapUpdateNeeded {
+			_, err = p.kubeClient.CoreV1().ConfigMaps(p.managerNamespace).Update(configMap)
```

Comment:
Don't we miss a retry here ?

Reply:
In this routine it is not needed, since if it fails then it will succeed the next 3 seconds, no?

Reply:
ack